SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1456 @@
+		}
+	}
+	foreach ($configuration in $productConfiguration.Default.PSObject.Properties) {
+		if ($WSAConfigTypeValues.psobject.properties.match($configuration.Name).Count) {
+			if ($WSAConfigTypeValues.$($configuration.Name) -eq "Default") {
+				$WSAConfigTypeValues.$($configuration.Name) = $configuration.Value
+			}
+		}
+	}
+	$standardConfigValue = New-Object -TypeName psobject
+	$standardConfigValue | Add-Member -notepropertyname "productType" -notepropertyvalue "WSA"
+	$standardConfigValue | Add-Member -notepropertyname "productAttributes" -notepropertyvalue $WSAConfigTypeValues
+	$ppmStandardConfigValues.Add($standardConfigValue)
+}
+Function checkRange() {
+	Param (
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$name,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [int]$value,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [int]$minRange,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [int]$maxRange,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [Bool]$required
+	)
+	if (($value -eq "Null") -and ($required -eq $true)) {
+		Write-LogMessage -Type ERROR -Message "$name variable has not been configured, please double check the standard configuration json file"  -Colour Red
+		return $false
+	} elseif (($value -lt $minRange) -or ($value -gt $maxRange)) {
+		Write-LogMessage -Type ERROR -Message "The recommended range for $name should be between $minRange and $maxRange"  -Colour Red
+		return $false
+	}
+	else {
+		return $true
+	}
+}
+Function checkEmailString() {
+	Param (
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$name,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$address,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [Bool]$required
+	)
+	
+	if (($address -eq "Null") -and ($required -eq $true)) {
+		Write-LogMessage -Type ERROR -Message "$name variable has not been configured, please double check the standard configuration json file" -Colour Red
+		return $false
+	}
+	$checkStatement = $address -match "^\w+([-+.']\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$"
+	if ($checkStatement -eq $true) {
+		return $true
+	} else {
+		Write-LogMessage -Type ERROR -Message "please input a validate email address for" $name "in the standard configuration json file"  -Colour Red
+		return $false
+	}
+}
+Function importStandardConfigurations {
+	<#
+		importing the standardconfiguration json into a consumable array to feed into various functions.
+	#>
+	Param (
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [psobject]$ppmVariables,
+		[Parameter (Mandatory = $true)] [AllowEmptyCollection()][System.Collections.Generic.List[System.Object]]$ppmStandardConfigValues,
+        [Parameter (Mandatory = $true)] [AllowEmptyCollection()][System.Collections.Generic.List[System.Object]]$ppmEnvironmentalDetails
```

Comment:
Formatting, all Parameters in line

Reply:
addressed in 9ee41934dd0ee5a302dd128339b72e2f82898175